misc/test_runner/include/ia2_test_runner.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+#include <ia2.h>
+#include <stdio.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <assert.h>
+
+#if !defined(IA2_TEST_RUNNER_SOURCE)
+typedef void *ia2_test_fn;
+#else
+typedef void (*ia2_test_fn)(void);
+#endif
+
+struct fake_criterion_test {
+  ia2_test_fn test;
+  ia2_test_fn init;
+  int exit_code;
+};
+
+#if !defined(IA2_TEST_RUNNER_SOURCE)
+#define Test(suite, name, ...)                                                                                                 \
+  IA2_BEGIN_NO_WRAP                                                                                                            \
+  void fake_criterion_##suite##_##name(void);                                                                                  \
+  IA2_END_NO_WRAP                                                                                                              \
+  __attribute__((__section__("fake_criterion_tests"))) struct fake_criterion_test fake_criterion_##suite##_##name##_##test = { \
+      .test = fake_criterion_##suite##_##name,                                                                                 \
+      ##__VA_ARGS__};                                                                                                          \
+  void fake_criterion_##suite##_##name(void)
+#endif
+
+#define cr_log_info(f, ...) printf(f "\n", ##__VA_ARGS__)
+#define cr_log_error(f, ...) fprintf(stderr, f "\n", ##__VA_ARGS__)
+
+#define cr_assert assert
+#define cr_assert_eq(a, b) cr_assert((a) == (b))
+#define cr_assert_lt(a, b) cr_assert((a) < (b))
+#define cr_fatal(s)          \
+  do {                       \
+    fprintf(stderr, s "\n"); \
+    exit(1);                 \
+  } while (0)
+
+/*
+ * This header defines a test framework for detecting MPK violations using
+ * signal handlers. This file must be included exactly once from a source file
+ * in the main binary with IA2_DEFINE_TEST_HANDLER defined by the preprocessor.
+ * This will define the functions and variables used by the test handler, ensure
+ * it is initialized before main and provide access to the LOG and
+ * CHECK_VIOLATION macros. Other files which need CHECK_VIOLATION or LOG may
+ * include the header without defining IA2_DEFINE_TEST_HANDLER. Using
```

Comment:
Is this still true? Do we need to define `IA2_DEFINE_TEST_HANDLER`?

Reply:
oh right, I forgot to update the comments

Reply:
updated comments